### ISSUE TITLE
Custom reverse proxy to Tor not working

### ISSUE THREAD

**Issue by danielcharrua** [btcpayserver/btcpayserver-doc#917]
Following the docs guide on creating a [reverse proxy to Tor](https://docs.btcpayserver.org/ReverseProxyToTor/) I get a blank page and error 400 after sending the login information. I hace strictly followed all steps mounting this on a DigitalOcean fresh droplet.

I've found that commenting out `proxy_set_header Connection "upgrade";` I can login to BTCPay Server but with the following errors on the JS console
```
WebSocket connection to 'wss://pay.xxx.es/Notifications/SubscribeUpdates' failed: Unexpected response code: 400
Error while connecting to websocket for notifications (callback)
```

**Comment by dennisreimann** [comment]
Does it work fine in Tor browser using the address provided as `REMOTE_HOST`?

**Comment by danielcharrua** [comment]
> Does it work fine in Tor browser using the address provided as `REMOTE_HOST`?

Yes, just tried it on a Tor browser and I can login without any issues and any console warnings/errors

**Comment by dennisreimann** [comment]
Do the nginx logs on both the remote host and the proxy provide any more information?

**Comment by dennisreimann** [comment]
> I can't see anything else. Has anyone have tried this setup?

I have it running in production and wrote the doc based on my setup. The connection upgrade is for the web sockets, but shouldn't lead to a general error 400.

**Comment by danielcharrua** [comment]
I see lots of people struggling with this in particular but any of them have a solution or workaround. I understand that the connection upgrade is for web sockets but in my case is leading to a 400 error and a blank page is shown. I have followed your steps and started again 3 times 😬

I have the BTCPay Server installed in a Raspi, under Umbrel. Maybe my router is in the way? This is possible?
The weird thing is that if I use a Tor browser all goes smooth... Is there anything else I can test to help debugging this?

**Comment by dennisreimann** [comment]
The router shouldn't be involved here, this is one of the main points of this setup. 
Did you try the reverse SSH as an alternative?

**Comment by danielcharrua** [comment]
Reverse SSH is not an alternative at this moment as I don't want to install any extra config on the Umbrel node. But I think I've found a way, just checking I've found this [http://nginx.org/en/docs/http/websocket.html](http://nginx.org/en/docs/http/websocket.html) and added to `/etc/nginx/conf.d/map.conf`

```
map $http_upgrade $connection_upgrade {
  default upgrade;
  ''      close;
}
```

And edited `/etc/nginx/sites-available/btcpayserver.conf` to:

```
location / {
    proxy_pass http://127.0.0.1:9081/;
    proxy_http_version 1.1;
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $remote_addr;
    proxy_set_header X-Forwarded-Proto $proxy_x_forwarded_proto;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $connection_upgrade;
  }
```

Seems to be working now, please tell me what do you think.

**Comment by dennisreimann** [comment]
I just checked and my config does not contain such a mapping. Nevertheless I added it and it still works fine.

Would you mind extending the doc with that part? I'll merge it in so that others don't stumble at that point as well. Thanks!

**Comment by danielcharrua** [comment]
#923 created

**Comment by dennisreimann** [comment]
Fixed via #923.
